### ISSUE TITLE
Windows' height resets to smaller screen height on workspace reopen (Linux/Gnome/Cinnamon + two screens)

### ISSUE THREAD

**Issue by lhalda** [microsoft/vscode#98362]
**VSCode Version:**
- Version: 1.45.1
- Commit: 5763d909d5f12fe19f215cbfdd29a91c0fa9208a
- Date: 2020-05-14T08:27:22.494Z
- Electron: 7.2.4
- Chrome: 78.0.3904.130
- Node.js: 12.8.1
- V8: 7.8.279.23-electron.0
- OS: Linux x64 5.0.0-32-generic

**System:**
- Linux Mint 19.3 Cinnamon
- Laptop screen (secondary): 1920x1080
- Big screen (external, primary): 2560x1440
- Zoom / DPI setting on both screens: Normal/100% (no zoom because of similar physical DPI)
- This has been an issue for me for months, I recently upgraded from Mint 19.2 to Mint 19.3 and reinstalled the computer, brought over the VSCode settings
- VSCode is installed from Microsoft's apt repository (`deb [arch=amd64 signed-by=/usr/share/keyrings/packages.microsoft.gpg] https://packages.microsoft.com/repos/vscode stable main`) and regularly updated

**Steps to Reproduce:**

1. Snap the VSCode windows with open workspaces to left and right halves of the big screen (external, primary screen)
2. Close VSCode by clicking File -> Exit (closes all windows)
3. Reopen VSCode (opens all previously closed windows)
4. The windows are in correct positions and with correct widths, but the heights are copying the smaller screen (1080 px). When I snap them back to edges, the side panel (Explorer) is still resized, so I need to manually increase height of Open Editors and Workspace in all windows, because after snapping (increasing height), only the last thing's height was extended (usually workspace, outline, or TODOs from plugin for me).

**Screenshot** after VSCode startup with extensions disabled; smaller than screen height:
![image](https://user-images.githubusercontent.com/2124851/82630639-78080e80-9c1d-11ea-8ff4-6daf38e19cca.png)
(In the screenshot, it really is Linux Mint with Cinnamon, but I like the way my usual Windows 10 workspace looks and works and tried to replicate it in my work computer where I need to use Linux.)

**Expected Behavior:** after reopening, all the windows keep same position and size as was when I exited VSCode.

I tested with `code --disable-extensions` to check (no extensions exit -> no extensions start), windows are still resized after VSCode starts.

**Comment by deepak1556** [comment]
Can you check if this issue happens with our [insiders version](https://code.visualstudio.com/insiders) and [exploration version](https://az764295.vo.msecnd.net/exploration/5103b1b09c0e26d466756bfdc1546a8c4bfd68c8/code-exploration_1.52.0-1604388902_amd64.deb) that come with newer runtime. Thanks!

**Comment by lhalda** [comment]
@deepak1556 Hello! I just tested these two versions:
code-insiders_1.52.0-1606110790_amd64.deb
code-exploration_1.52.0-1604388902_amd64.deb

Both behave the same - the windows get resized to be shorter vertically, but remain the same horizontally. I tried it without any changed settings and for insiders version also with two of my workspaces open, the results are always the same.

There is a difference from my regular vscode (haven't updated in a month, still on version 1.50.1 - but the issue hasn't changed for about a year now). I think it is because of different settings. Anyway, in my regular vscode, the resizing happens for height, but the width and position of top left and top right corners of the windows remain the same. In both versions tested (insiders and exploration), without changing any settings, the position of both windows (snapped to the left and to the right edge of the screen) goes lower by about the title bar height each sub-sequential restart and the window snapped to the screen's left half moves slightly to the right (I'd say about a pixel or two each restart), while the right window stays the same horizontally.

Thanks for your attention :) Please let me know if you need me to elaborate more.

**Comment by lhalda** [comment]
I recorded the behavior of insiders and exploration version. I didn't record the stable version because I didn't want to accidentally show my projects.

Outline of my screens (left is laptop screen 1080p - secondary, right is external screen 1440p - primary):
![screens](https://user-images.githubusercontent.com/2124851/100717094-333f0a80-33ec-11eb-9c06-f8999163babe.png)


Video (primary screen only):
https://drive.google.com/file/d/17KiBuOV00TqTImYiW3TMa0ZBPX4L6JYk/view


As I stated in the initial post, I made the system look like Windows 10 as much as I could, but actually it is Linux Mint with Cinnamon. The recording was done today after apt upgrade of everything.

System:
Linux Mint 19.3 Cinnamon
Cinnamon 4.4.8
Linux Kernel 5.0.0-32-generic
GPU 1: Intel Corporation 3rd Gen Core processor Graphics Controller
GPU 2: NVIDIA Corporation GK107GLM [Quadro K2000M]
